GCC 9.1.1 strict-overflow warnings

GCC warns when enabling ```-Wstrict-overflow=n``` for ```n > 1```. Some of the warnings can be suppressed with ```#pragma```s, but others can't as they are triggered at the call site during ```constexpr``` expansion. 

```
[build] .../frozen-master/include/frozen/bits/pmh.h:174:8: error: assuming pointer wraparound does not occur when comparing P +- C1 with P +- C2 [-Werror=strict-overflow]
[build]   174 |   auto buckets = step_one.get_sorted_buckets();
[build]       |        ^~~~~~~
```

```
[build] .../frozen-master/include/frozen/bits/pmh.h:228:1: error: assuming pointer wraparound does not occur when comparing P +- C1 with P +- C2 [-Werror=strict-overflow]
[build]   228 | }
[build]       | ^
```

--- a/include/frozen/bits/algorithms.h
+++ b/include/frozen/bits/algorithms.h
@@ -112,7 +112,7 @@ constexpr Iterator partition(Iterator left, Iterator right, Compare const &compa
   auto pivot = left + (right - left) / 2;
   auto value = *pivot;
   cswap(*right, *pivot);
-  for (auto it = left; it < right; ++it) {
+  for (auto it = left; 0 < right - it; ++it) {
     if (compare(*it, value)) {
       cswap(*it, *left);
       left++;
@@ -124,7 +124,7 @@ constexpr Iterator partition(Iterator left, Iterator right, Compare const &compa
 
 template <typename Iterator, class Compare>
 constexpr void quicksort(Iterator left, Iterator right, Compare const &compare) {
-  while (left < right) {
+  while (0 < right - left) {
     auto new_pivot = bits::partition(left, right, compare);
     quicksort(left, new_pivot, compare);
     left = new_pivot + 1;

